### PR TITLE
Add LittEleven

### PR DESCRIPTION
**Pull request by Spandamn**
Halp.

### CODE REVIEW

File: data/mods/ssb/moves.ts [hunk 1]
```diff
@@ -1083,6 +1083,77 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		category: "Status",
+		desc: "If this Pokemon does not take damage this turn, it switches out to another Pokemon in the party and gives it the Download boost. Fails otherwise.",
+		shortDesc: "If not damaged this turn, switches out and gives next Pokemon Download boost; fails otherwise.",
+		name: "/nexthunt",
+		pp: 10,
+		priority: 0,
+		flags: {protect: 1, mirror: 1},
+		sideCondition: "nexthunt",
+		beforeTurnCallback(pokemon) {
+			pokemon.side.addSideCondition("nexthunt");
```
Comment (Spandamn) [inline data/mods/ssb/moves.ts]: Adds the side condition as soon as the pokemon switchess in, hopefully.

File: data/mods/ssb/moves.ts [hunk 2]
```diff
@@ -1083,6 +1083,77 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Teleport', source);
+		},
+		effect: {
+			onStart(side, source) {
+				if (side.sideConditions.duration === 0) return;
```
Comment (Spandamn) [inline data/mods/ssb/moves.ts]: Needs a check for when this is called from onTryHit

File: data/mods/ssb/abilities.ts [hunk 3]
```diff
@@ -515,6 +515,42 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
...
+			if ((source.side === dazzlingHolder.side || move.target === 'all') && move.priority > 0.1) {
+				this.attrLastMove('[still]');
+				this.add('cant', dazzlingHolder, 'ability: Dark Royalty', move, '[of] ' + target);
+				return false;
+			}
+		},
+		onModifyAtkPriority: 5,
+		onModifyAtk(atk, attacker, defender, move) {
+			if (move.type === 'Dark') {
+				this.debug('Dark Royyalty boost');
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts]:
```suggestion
				this.debug('Dark Royalty boost');
```

File: data/mods/ssb/moves.ts [hunk 4]
```diff
@@ -1083,6 +1083,77 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+				this.effectData.positions = [];
+				for (const i of side.active.keys()) {
+					this.effectData.positions[i] = false;
+				}
+				this.effectData.positions[source.position] = true;
+			},
+			onDamage(damage, target, source, move) {
+				if (target === source) return;
+				if (damage > 0 && target.side === this.effectData.target && target === this.effectData.source) {
+					target.side.removeSideCondition("nexthunt");
```
Comment (Spandamn) [inline data/mods/ssb/moves.ts]: If damaged, no longer need the side condition. removes it.

File: data/mods/ssb/abilities.ts [hunk 5]
```diff
@@ -515,6 +515,42 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
...
+			}
+		},
+		onModifySpAPriority: 5,
+		onModifySpA(atk, attacker, defender, move) {
+			if (move.type === 'Dark') {
+				this.debug('Dark Royyalty boost');
+				return this.chainModify(1.2);
+			}
+		},
+		name: "Queenly Majesty",
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts]:
```suggestion
		name: "Dark Royalty",
```

File: data/mods/ssb/abilities.ts [hunk 6]
```diff
@@ -515,6 +515,42 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
...
+		rating: 2,
+		num: 214,
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts]:
```suggestion
```

File: data/mods/ssb/moves.ts [hunk 7]
```diff
@@ -1083,6 +1083,77 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+					}
+					positions[target.position] = false;
+				}
+				if (!positions.some(affected => affected === true)) {
+					target.side.removeSideCondition('nexthunt');
+				}
+			},
+		},
+		onTryHit(pokemon) {
+			if (!pokemon.side.sideConditions.nexthunt) return;
```
Comment (Spandamn) [inline data/mods/ssb/moves.ts]: Fails if side condition was removed.

File: data/mods/ssb/moves.ts [hunk 8]
```diff
@@ -1083,6 +1083,77 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		desc: "If this Pokemon does not take damage this turn, it switches out to another Pokemon in the party and gives it the Download boost. Fails otherwise.",
+		shortDesc: "If not damaged this turn, switches out and gives next Pokemon Download boost; fails otherwise.",
+		name: "/nexthunt",
+		pp: 10,
+		priority: 0,
+		flags: {protect: 1, mirror: 1},
+		sideCondition: "nexthunt",
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts:1097]:
```suggestion
		sideCondition: 'nexthunt',
```

File: data/mods/ssb/moves.ts (hunk 1 above)
Comment (KrisXV) [inline data/mods/ssb/moves.ts]:
```suggestion
			pokemon.side.addSideCondition('nexthunt');
```

File: data/mods/ssb/random-teams.ts [hunk 9]
```diff
@@ -204,6 +204,12 @@ export const ssbSets: SSBSets = {
 		signatureMove: 'Alphabet Soup',
 		evs: {spa: 252, spd: 4, spe: 252}, ivs: {atk: 0}, nature: 'Timid',
 	},
+	"Litt♥Eleven": {
+		species: 'Bisharp', ability: 'Dark Royalty', item: 'Black Glasses', gender: 'M',
+		moves: ['Sucker Punch', 'Knock Off', 'Iron Head'],
+		signatureMove: ' /nexthunt',
```
Comment (KrisXV) [inline data/mods/ssb/random-teams.ts]:
```suggestion
		signatureMove: '/nexthunt',
```

File: data/mods/ssb/moves.ts (hunk 8 above)
Comment (Spandamn) [inline data/mods/ssb/moves.ts:1097]: will set side condition after move starts.

File: data/mods/ssb/moves.ts [hunk 10]
```diff
@@ -1083,6 +1083,77 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+				}
+				if (!positions.some(affected => affected === true)) {
+					target.side.removeSideCondition('nexthunt');
+				}
+			},
+		},
+		onTryHit(pokemon) {
+			if (!pokemon.side.sideConditions.nexthunt) return;
+		},
+		selfSwitch: false,
```
Comment (Spandamn) [inline data/mods/ssb/moves.ts]: need to make this true if no damage was taken

File: data/mods/ssb/moves.ts [hunk 11]
```diff
@@ -1083,6 +1083,58 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+			this.add('-anim', source, 'Teleport', source);
+		},
+		onTryHit(target, source, move) {
+			if (!move.selfSwitch) return;
+		},
+		effect: {
+			duration: 1,
+			onHit(pokemon, source, move) {
+				if (move.category !== 'Status' && pokemon === this.effectData.source) {
+					delete move.selfSwitch;
```
Comment (xInstruct) [inline data/mods/ssb/moves.ts:1111]:
```suggestion
					this.effectData.source.side.removeSideCondition("nexthunt");
					
```

File: data/mods/ssb/abilities.ts [hunk 12]
```diff
@@ -515,6 +515,40 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 		},
 	},
 
+	// LittEleven
+	darkroyalty: {
+		desc: "While this Pokemon is active, priority moves from opposing Pokemon targeted at allies are prevented from having an effect.",
+		shortDesc: "While this Pokemon is active, allies are protected from opposing priority moves.",
```
Comment (ghost) [inline data/mods/ssb/abilities.ts]: that isnt all of the ability

File: data/mods/ssb/moves.ts [hunk 13]
```diff
@@ -1083,6 +1083,58 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Teleport', source);
+		},
+		onTryHit(target, source, move) {
+			if (!move.selfSwitch) return;
+		},
+		effect: {
+			duration: 1,
+			onHit(pokemon, source, move) {
```
Comment (ghost) [inline data/mods/ssb/moves.ts:1109]: So, this is going to activate regardless because its not looking for when damage is taken before the turn starts

Comment (ghost) [reply data/mods/ssb/moves.ts:1109]: I think the best way to do this is to move all this into a different side condition under tracker in statuses.js, then make a clone of the focuspunch volatile and apply it here. Or vise versa idc

**Pull request closed without merging**